src/types/Map.ts

```diff
@@ -131,6 +131,10 @@ export type MapData = Record<MapObject, MapSourceBaseData | undefined>;
 export type MapControl = {
   // hide the full screen control
   hideFullScreen?: boolean;
+
+  // custom map controls
+  topRightMapControl?: React.ReactNode;
+  bottomLeftMapControl?: React.ReactNode;
```

Comment:
add the controls to MapControl properties so we don't have to re-define them across map component properties passing